src/v/storage/disk_log_impl.cc

```diff
@@ -579,6 +587,23 @@ ss::future<bool> disk_log_impl::sliding_window_compact(
           result);
         has_self_compacted = true;
     }
+
+    // In the case that all segments in the range were already cleanly
+    // compacted, we allowed self-compaction to remove all tombstones in the
+    // previous step. However, it would be a no-op to perform window compaction
+    // on the segments past this point.
+    const bool all_clean_compacted = std::ranges::all_of(
```

Comment:
Any good way to deduplicate repeated `all_clean_compacted` code between here and `find_sliding_range()`?

Reply:
Helper method in an anonymous namespace?

Reply:
Removed dead code.